.circleci/config.yml

```diff
@@ -9,149 +9,273 @@ jobs:
       LANG: C.UTF-8
     steps:
       - checkout
-      - run: pip3 install flake8
-      - run: flake8 --exclude "" --ignore=E501,F821 geotrek/settings
-      - run: flake8 geotrek
-      - run: test $(ls geotrek/*/migrations/*.py | xargs grep -l srid | xargs grep -L SRID | wc -l) -eq 0
+      - run:
+          name: Install flake8
+          command: pip3 install flake8
+      - run:
+          name: Lint with flake8
+          command: |
+            flake8 --exclude "" --ignore=E501,F821 geotrek/settings
+            flake8 geotrek
+      - run:
+          name: Check no SRID value in migration files
+          command: test $(ls geotrek/*/migrations/*.py | xargs grep -l srid | xargs grep -L SRID | wc -l) -eq 0
+
   doc:
     docker:
       - image: python:3.6
     environment:
       LANG: C.UTF-8
+    working_directory: ~/project/docs
     steps:
       - checkout
-      - run: |
-          cd docs/
-          pip install -r ./requirements.txt
-          make html
+      - run:
+          name: Install Sphinx
+          command: |
+            cd docs/
+            pip install -r ./requirements.txt
+      - run:
+          name: Build documentation
+          command: |
+            cd docs/
+            make html
+
 
   docker_image:
     machine:
       image: ubuntu-2004:202201-02
     steps:
       - checkout
-      - run: BASE_IMAGE_TAG=focal-3.8 make build
-      - run: echo "$DOCKER_PASSWORD" | docker login --username "$DOCKER_LOGIN" --password-stdin
-      - run: docker tag geotrek:latest geotrekce/circleci-admin:$CIRCLE_WORKFLOW_ID
-      - run: docker push geotrekce/circleci-admin:$CIRCLE_WORKFLOW_ID
+      - run:
+          name: Build docker image base on Ubuntu 20.04
+          command: BASE_IMAGE_TAG=focal-3.8 make build
+      - run:
+          name: Tag and save docker image
+          command: |
+            docker tag geotrek:latest geotrekce/circleci-admin:$CIRCLE_WORKFLOW_ID
+            docker save -o ./20.04.$CIRCLE_WORKFLOW_ID.tar geotrekce/circleci-admin:$CIRCLE_WORKFLOW_ID
+      - persist_to_workspace:
+          root: .
+          paths:
+            - 20.04.*.tar
```

Comment:
Pourquoi tu reutilises pas $CIRCLE_WORKFLOW_ID ?

Reply:
no matter WORKFLOW ID because persist_to_workspace is unique by workflow

Reply:
and it's not possible to use ENV variable here

Reply:
fixed by using $CIRCLE_SHA1 (allow to relaunch failed job) and not specify ID in saved image